numpy/ma/core.py

```diff
@@ -3029,18 +3029,20 @@ def __array_wrap__(self, obj, context=None):
 
         if context is not None:
             result._mask = result._mask.copy()
-            (func, args, _) = context
-            m = reduce(mask_or, [getmaskarray(arg) for arg in args])
+            func, args, out_i = context
+            # args sometimes contains outputs (gh-10459), which we don't want
+            input_args = args[:func.nin]
+            m = reduce(mask_or, [getmaskarray(arg) for arg in input_args])
             # Get the domain mask
             domain = ufunc_domain.get(func, None)
             if domain is not None:
                 # Take the domain, and make sure it's a ndarray
-                if len(args) > 2:
+                if len(input_args) > 2:
```

Comment:
Not directly related, but this is very strange -- there are no ufuncs with more than two input arguments, so this path will never get taken... And before this would only be taken if there was an output argument, for which the domain would then be checked. Which makes no sense at all!

Reply:
> there are no ufuncs with more than two input arguments, so this path will never get taken

There's nothing to stop a third party (or scipy?) adding one like this, so it's nice to try to support it.

Reply:
Then they also have to update the domain dict -- not impossible, I guess, but why are they then forced to calculate the domain by reduction rather than just by getting all the arguments? This is definitely not what I would expect... for most operations, the masking depends very differently on each argument (think division...).

Reply:
You're right, this doesn't make sense as is - expanding the tuple arguments is the right thing to do every time. I can fix this much later today, either in this PR or a separate one.

Doing it in a separate one means that I can also rebase the `__array_prepare__` PR on top of this tonight.